[rocprofiler-compute] Update Documentation for Strix Halo (#4794)
ROCm/rocm-systems: projects/rocprofiler-compute/docs/conceptual/rdna/rdna-performance-model.rst

   :align: center

   **Figure: AMD RDNA3 generation series (block diagram).** See page 5 of
   `RDNA3 shader instruction set architecture <https://docs.amd.com/v/u/en-US/rdna3-shader-instruction-set-architecture-feb-2023_0#page=5>`__.

For Instinct / CDNA naming (CU, shader engine, etc.), use the top-level :doc:`../performance-model`
overview and :doc:`../cdna/cdna-performance-model`. Here the focus is on WGPs, TCP / GL1C / GL2C,
GCEA, and related panels when an analysis config targets RDNA hardware.

Public architecture summaries and :doc:`GPU / accelerator specifications <rocm:reference/gpu-arch-specs>`
remain the best reference for packaging, SIMD width, and generational differences between
RDNA3, RDNA3.5, and later APUs. The sections below describe what the profiler measures and names
for RDNA3.5 (gfx1151).

ROCm Profiler includes analysis panels targeting RDNA3.5 parts reporting as
gfx1151 — for example integrated graphics on AMD Ryzen AI Max Series - Strix Halo
processors.

.. rubric:: Memory hierarchy in the tool

For gfx1151, the Memory Chart panel walks the path from instruction and scalar

> paths, TCP (GL0), LDS, interfaces to GL1C (L1), GL2C (L2), and GCEA toward

Reviewer: Can we remove the (L1) (L2) brackets here? Technically we should be using GL0, GL1, GL2, to specify that it is a global Lx cache, not just a level cache. Seems redundant to put "GL1C (GL1)"

Reviewer 2: removed in commit "c7fabbae90598be84ccaf5441ea6973c8d91a6b1"

system memory.

.. rubric:: Workgroups and execution

RDNA3-class GPUs organize compute around Workgroup Processors (WGPs) and
Compute Units (CUs); wavefronts are typically wave32-oriented in this
configuration. The WGP, SPI, and CPC panels in gfx1151 expose
dispatch, occupancy, and command-processor side metrics that complement the
:doc:`Instinct / CDNA conceptual pages <../cdna/compute-unit>` (which use
naming such as CU and SE in places).

.. rubric:: Where to read metric text

* Conceptual sub-pages below group metric tables by block (System Speed-of-Light
  first, then WGP, TCP (GL0), GL1 / GL1C, GL2 / GL2C + GCEA, shader engine / SPI,
  command processor / CPC). The :doc:`system-speed-of-light` page uses the same
  metric keys as the analysis panel.

In this chapter, profiler concepts for RDNA use APU naming (WGP, GL1/GL2, ...)
and embed the RDNA3.5 (gfx1151) metric tables under each block:

* :doc:`system-speed-of-light` — SoL table for gfx1151.

* :doc:`wgp` — roofline, WGP utilization, waves, instruction mix, WGP instruction and data caches.

* :doc:`tcp-cache` — TCP (GL0): panel tables and Memory Chart rows through TCP-GL1.

> * :doc:`gl1-cache` — GL1C (L1): panel tables and Memory Chart GL1C Cache (L1).

Reviewer: Same L1 -> GL1 point here

Reviewer 2: removed in commit "c7fabbae90598be84ccaf5441ea6973c8d91a6b1"

.. meta::

>    :description: ROCm Compute Profiler RDNA3 APU performance model

Reviewer 3: ```suggestion
   :description: Understand the AMD RDNA architectures and hardware blocks with ROCm Compute Profiler to analyze and optimize performance on AMD Ryzen APUs.
```

Reviewer 2: resolved in 206673250825b2c00edf3c245b5c52b2737ff869

>    :keywords: ROCm Compute Profiler, RDNA, RDNA3, gfx1151, Radeon, ROCm

Reviewer 3: ```suggestion
   :keywords: ROCm Compute Profiler, RDNA, RDNA3, gfx1151, Radeon, ROCm, Ryzen
```

Reviewer 2: resolved in 206673250825b2c00edf3c245b5c52b2737ff869

.. _rdna-performance-model:

=====
RDNA3
=====

> This chapter covers AMD Radeon / RDNA configurations exposed in ROCm Compute Profiler.

Reviewer 3: ```suggestion
This chapter covers AMD Ryzen / RDNA configurations exposed in ROCm Compute Profiler.
```

Reviewer 2: resolved in 206673250825b2c00edf3c245b5c52b2737ff869

.. figure:: ../../data/conceptual/RDNA3_Block_Diagram.png
   :alt: AMD RDNA3 generation series block diagram — host CPU, system and device memory, memory controller, L2/L1 caches, global data share, command processors, ultra-threaded dispatch, and processor array of WGPs with CUs, LDS, instruction and constant caches
   :align: center

   **Figure: AMD RDNA3 generation series (block diagram).** Work flows from the host and from
   system/device memory through the memory controller into the L2 / L1 stack, Global Data

>    Share, and command processors, then through the Ultra-Threaded Dispatch Processor into a

Reviewer: "ultra-threaded dispatch processor" is an extremely old concept linked to as far back as R600. Is there a term for latest architecture?

For Instinct / CDNA naming (CU, shader engine, etc.), use the top-level :doc:`../performance-model`

> overview and :doc:`../cdna/cdna-performance-model`. Here the focus is on WGPs, TCP / GL1C / GL2C,

Reviewer 4: "For Instinct / CDNA naming (CU, shader engine, etc.), use the top-level :doc:`../performance-model`
overview and :doc:`../cdna/cdna-performance-model`." put this into a Note widget.

"Here the focus is on WGPs, TCP / GL1C / GL2C,
GCEA, and related panels when an analysis config targets RDNA hardware." is implied with the title "RDNA3".

Reviewer 2: resolved in 11f31a720ac0d18e3199c1a0458db2be8a5903cd

GCEA, and related panels when an analysis config targets RDNA hardware.

Public architecture summaries and :doc:`GPU / accelerator specifications <rocm:reference/gpu-arch-specs>`
remain the best reference for packaging, SIMD width, and generational differences between
RDNA3, RDNA3.5, and later APUs. The sections below describe what the profiler measures and names
for RDNA3.5 (gfx1151).

> ROCm Profiler includes analysis panels targeting RDNA3.5 parts reporting as

Reviewer 4: ```suggestion
ROCm Compute Profiler includes analysis panels targeting RDNA3.5 parts reporting as
```

> GCEA, and related panels when an analysis config targets RDNA-class parts.

Reviewer: ```suggestion
GCEA, and related panels when an analysis config targets RDNA hardware.
```
or anything to change "class parts"

> gfx1151 — for example integrated graphics on AMD Ryzen AI Max Series - Strix Halo

Reviewer 4: ```suggestion
gfx1151 — for example, integrated graphics on AMD Ryzen AI Max Series - Strix Halo
```

RDNA3, RDNA3.5, and later client GPUs. The sections below describe what the profiler measures and names
for RDNA3.5 (gfx1151).

ROCm Profiler includes analysis panels targeting RDNA3.5 parts reporting as

> gfx1151 — for example integrated graphics on AMD Ryzen AI Max+ (Strix Halo) class

Reviewer: NEEDED CHANGED EVERYWHERE:
We need to modify this to say AMD Ryzen AI Max Series - strix halo and box is not exclusive to the Max+
Support is 385, 390, and 395 (the latter is the only Max+)

Reviewer 2: Have changed it in commit "e274912be85d5593384345cb10cf457d7db8ce94"

processors.

.. rubric:: Memory hierarchy in the tool

For gfx1151, the Memory Chart panel walks the path from instruction and scalar
paths, TCP (vector L0), LDS, interfaces to GL1C (L1), GL2C (L2), and GCEA toward
system memory.

.. rubric:: Workgroups and execution

RDNA3-class GPUs organize compute around Workgroup Processors (WGPs) and
Compute Units (CUs); wavefronts are typically wave32-oriented in this
configuration. The WGP, SPI, and CPC panels in gfx1151 expose
dispatch, occupancy, and command-processor side metrics that complement the
:doc:`Instinct / CDNA conceptual pages <../cdna/compute-unit>` (which use
naming such as CU and SE in places).

.. rubric:: Where to read metric text

* Conceptual sub-pages below group metric tables by block (System Speed-of-Light
  first, then WGP, TCP (GL0), GL1 / GL1C, GL2 / GL2C + GCEA, shader engine / SPI,
  command processor / CPC). The :doc:`system-speed-of-light` page uses the same
  metric keys as the analysis panel.

In this chapter, profiler concepts for RDNA use client GPU naming (WGP, GL1/GL2, ...)
and embed the RDNA3.5 (gfx1151) metric tables under each block:

* :doc:`system-speed-of-light` — SoL table for gfx1151.

> * :doc:`wgp` — roofline, WGP utilization, waves, instruction mix, WGP I$/scalar caches.

Reviewer: ```suggestion
* :doc:`wgp` — roofline, WGP utilization, waves, instruction mix, WGP I$ and scalar caches.
```

Separate scalar caches phrase from I$ which is the sequencer cache or instruction cache

Reviewer: Also using the dollar sign and wording interchangeably is a bit odd, maybe "instruction and data cache" is a better suggestion!

```suggestion
* :doc:`wgp` — roofline, WGP utilization, waves, instruction mix, WGP instruction and data caches.
```

* :doc:`gl2-cache` — GL2C, GCEA / DRAM / arbiter, and related panel metrics.

* :doc:`shader-engine` — GRBM GPU/SE utilization and SPI dispatch statistics.

* :doc:`command-processor` — CPC / Micro Engine (ME) metrics (same role as CDNA CP, different tab layout in gfx1151).

> * :doc:`references` — public references and link to Instinct citations.

Reviewer 4: Not critical for this PR, but this should be organized into a table for RDNA vs CDNA naming conventions.

Reviewer 2: will work in later pr

Note
====

> rocprof-compute currently has limited support for WMMA on Strix Halo.

Reviewer 4: ```suggestion
ROCm Compute Profiler currently has limited support for WMMA on Strix Halo.
```
